rpcclient/local_test.go

```diff
@@ -225,11 +246,15 @@ func TestMempool2Nodes(t *testing.T) {
 	local := NewLocal(node1)
 	require.NotNil(local)
 
-	resp, err := local.BroadcastTxSync(ctx, []byte("foobar"))
+	// broadcast the bad Tx, this should not be propogated or added to the local mempool
+	resp, err := local.BroadcastTxSync(ctx, []byte("bad"))
+	assert.NoError(err)
+	assert.NotNil(resp)
+	// broadcast the bad Tx, this should not be propogated or added to the local mempool
+	resp, err = local.BroadcastTxSync(ctx, []byte("good"))
 	assert.NoError(err)
 	assert.NotNil(resp)
-
 	time.Sleep(1 * time.Second)
 
-	assert.Equal(node2.Mempool.TxsBytes(), int64(len("foobar")))
+	assert.Equal(node2.Mempool.TxsBytes(), int64(len("good")))
```

Comment:
modified the test that checks that the mempool is working to also check that invalid transactions do not get gossiped.